src/libraries/Common/tests/System/Net/Sockets/SocketTestExtensions.cs

```diff
@@ -38,14 +38,21 @@ public static void ForceNonBlocking(this Socket socket, bool force)
             }
         }
 
-        public static (Socket, Socket) CreateConnectedSocketPair()
+        public static (Socket client, Socket server) CreateConnectedSocketPair(bool ipv6 = false, bool dualModeClient = false)
```

Comment:
Nit: these are not `client` and `server`, but two connected TCP peers. The `server` Socket doesn't leave the method, and it looks like we are not Disposing it, which we probably should.

Reply:
> Nit: these are not client and server, but two connected TCP peers. 

The terminology reflects their roles during the establishment of the connection. I understand that this information is not needed afterwards, but still I find the labels useful for convenience, because it helps me mentally map the tests to a common real-world scenarios. For example `dualModeClient = false` refers to the first `Socket client`. and it sounds more natural than `bool dualModeSocket1`, which would be the case if we'd rename the tuple. We use this terminology in other tests, although it's not necessary or meaningful to label anything as client/server.

> The `server` Socket doesn't leave the method, and it looks like we are not Disposing it, which we probably should.

You mean the `listner` I guess. I also don't like this, but this problem is not specific to the PR. We should fix it on all call sites which I would do separately.